src/main/java/com/google/googleidentity/oauth2/validator/AuthorizationEndpointRequestValidator.java

```diff
@@ -0,0 +1,145 @@
+            ClientDetailsService clientDetailsService) throws InvalidRequestException {
+        String clientID = request.getParameter(OAuth2ParameterNames.CLIENT_ID);
+
+        if (Strings.isNullOrEmpty(clientID)) {
+            throw new InvalidRequestException(InvalidRequestException.ErrorCode.NO_CLIENT_ID);
+        }
+
+        Optional<ClientDetails> client = clientDetailsService.getClientByID(clientID);
+
+        if (!client.isPresent()) {
+            throw new InvalidRequestException(
+                    InvalidRequestException.ErrorCode.NONEXISTENT_CLIENT_ID);
+        }
+
+        String redirectUri = request.getParameter(OAuth2ParameterNames.REDIRECT_URI);
+
+        if (Strings.isNullOrEmpty(redirectUri)) {
+            throw new InvalidRequestException(InvalidRequestException.ErrorCode.NO_REDIRECT_URI);
+        }
+
+        try {
+            redirectUri = URLDecoder.decode(redirectUri, "utf-8");
+        } catch (UnsupportedEncodingException e) {
+            log.log(Level.INFO, "Uri decode failed", e);
+        }
+
+        if (!OAuth2Utils.matchUri(client.get().getRedirectUrisList(), redirectUri)) {
+            throw new InvalidRequestException(
+                    InvalidRequestException.ErrorCode.REDIRECT_URI_MISMATCH);
+        }
+    }
+
+    /**
+     * Check whether the Get request is valid in authorization endpoint,
+     * Must be call after validateRedirectUri function
+     */
+    public static void validateGET(
+            HttpServletRequest request,
+            ClientDetailsService clientDetailsService) throws OAuth2Exception {
+
+        //here the clientID has been checked in validateRedirectUri function
+        String clientID = request.getParameter(OAuth2ParameterNames.CLIENT_ID);
+
+        ClientDetails client = clientDetailsService.getClientByID(clientID).get();
+
+        String responseType = request.getParameter(OAuth2ParameterNames.RESPONSE_TYPE);
+
+        if (Strings.isNullOrEmpty(responseType)) {
+            throw new InvalidRequestException(InvalidRequestException.ErrorCode.NO_RESPONSE_TYPE);
+        } else if (!responseType.equals("token") && !responseType.equals("code")) {
+            throw new UnsupportedResponseTypeException();
+        }
+
+        String grantType = responseType.equals("token") ? "implicit" : "authorization_code";
+
+        if (!client.getGrantTypesList().contains(grantType)) {
+            throw new UnauthorizedClientException();
+        }
+
+        if (!Strings.isNullOrEmpty(request.getParameter(OAuth2ParameterNames.SCOPE))
+                && client.getIsScoped()) {
+            Set<String> scope =
+                    OAuth2Utils.parseScope(request.getParameter(OAuth2ParameterNames.SCOPE));
+            if (!client.getScopesList().containsAll(scope)) {
+                throw new InvalidScopeException();
+            }
+        }
+    }
+
+    /**
+     * Check whether the Post request is valid in authorization endpoint
```

Comment:
nit: Add some comment to clarify that this isn't a request defined in the OAuth 2.0 spec RFC 6749, but handles user consent.